### PR TITLE
Issues/avoid databind

### PR DESCRIPTION
**Pull request by pmlopes**
Motivation:

Vert.x dependency on `jackson-databind` is optional, this PR ensures that it remains so. For this reason all yaml reads are done without databind, but using snake yaml directly with the safe constructor option that prevents the use of gadgets/serialization of dynamic mapping

### CODE REVIEW

File: src/test/java/io/vertx/openapi/UtilsTest.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import io.vertx.junit5.Timeout;
 import io.vertx.junit5.VertxExtension;
 import io.vertx.junit5.VertxTestContext;
+import io.vertx.openapi.impl.Utils;
```
Comment (pk-work) [inline src/test/java/io/vertx/openapi/UtilsTest.java:21]: UtilsTest, should also be moved into impl package

File: src/main/java/io/vertx/openapi/impl/Utils.java [hunk 2]
```diff
@@ -50,14 +49,36 @@ public static Future<JsonObject> readYamlOrJson(Vertx vertx, String path) {
         return succeededFuture(buff.toJsonObject());
       } else if ("yaml".equals(suffix) || "yml".equals(suffix)) {
         try {
-          JsonNode node = YAML_MAPPER.readTree(buff.toString());
-          return succeededFuture(new JsonObject(node.toString()));
-        } catch (JsonProcessingException e) {
+          final Yaml yaml = new Yaml(new OpenAPIYamlConstructor());
```
Comment (pk-work) [inline src/main/java/io/vertx/openapi/impl/Utils.java:52]: yaml could be a static field, in my opinion there is no need to instantiate it every time.

**Comment by pmlopes** [comment]
Wrong branch

File: src/main/java/io/vertx/openapi/impl/Utils.java [hunk 3]
```diff
@@ -50,14 +49,36 @@ public static Future<JsonObject> readYamlOrJson(Vertx vertx, String path) {
...
   }
+
+  /**
+   * Yaml allows map keys of type object, however json always requires key as String,
+   * this helper method will ensure we adapt keys to the right type
+   *
+   * @param yaml yaml map
+   * @return json map
+   */
+  private static Map<String, Object> jsonify(Map<Object, Object> yaml) {
```
Comment (pk-work) [inline src/main/java/io/vertx/openapi/impl/Utils.java:71]: It is tested indirectly, so it's okay to skip the test

**Review by pk-work — CHANGES_REQUESTED** [review]

**Pull request closed without merging**